tests/base/src/test/java/org/keycloak/tests/admin/SMTPConnectionTest.java

```diff
@@ -15,53 +15,65 @@
-import org.keycloak.representations.idm.UserRepresentation;
-import org.keycloak.testsuite.AbstractKeycloakTest;
-import org.keycloak.testsuite.util.GreenMailRule;
-import org.keycloak.testsuite.util.UserBuilder;
 
 import jakarta.mail.internet.MimeMessage;
 import jakarta.ws.rs.core.Response;
+import org.keycloak.testframework.annotations.InjectAdminClient;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.mail.MailServer;
+import org.keycloak.testframework.mail.annotations.InjectMailServer;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.RealmConfig;
+import org.keycloak.testframework.realm.RealmConfigBuilder;
+
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
 import static org.keycloak.representations.idm.ComponentRepresentation.SECRET_VALUE;
 
 /**
  * @author <a href="mailto:bruno@abstractj.org">Bruno Oliveira</a>
  */
-public class SMTPConnectionTest extends AbstractKeycloakTest {
+@KeycloakIntegrationTest
+public class SMTPConnectionTest {
 
-    public final String SMTP_PASSWORD = setSmtpPassword();
+    @InjectRealm(config = SMTPRealmWithClientAndUser.class)
+    private ManagedRealm managedRealm;
 
-    @Rule
-    public GreenMailRule greenMailRule = new GreenMailRule();
-    private RealmResource realm;
+    @InjectAdminClient(mode = InjectAdminClient.Mode.MANAGED_REALM, client = "myclient", user = "myadmin")
+    private Keycloak adminClient;
 
-    @Override
-    public void addTestRealms(List<RealmRepresentation> testRealms) {
-    }
+    @InjectMailServer
+    private MailServer mailServer;
 
-    public String setSmtpPassword() {
-        return "admin";
-    }
+    public static class SMTPRealmWithClientAndUser implements RealmConfig {
+
+        @Override
+        public RealmConfigBuilder configure(RealmConfigBuilder realm) {
+            realm.addClient("myclient")
+                    .secret("mysecret")
+                    .directAccessGrants();
 
-    @Before
-    public void before() {
-        realm = adminClient.realm("master");
-        List<UserRepresentation> admin = realm.users().search("admin", 0, 1);
-        UserRepresentation user = UserBuilder.edit(admin.get(0)).email("admin@localhost").build();
-        realm.users().get(user.getId()).update(user);
+            realm.addUser("myadmin")
+                    .name("My", "Admin")
+                    .email("admin@localhost")
+                    .emailVerified()
+                    .password("myadmin")
+                    .clientRoles(Constants.REALM_MANAGEMENT_CLIENT_ID, AdminRoles.REALM_ADMIN);
+
+            return realm;
+        }
```

Comment:
I would personally prefer putting the config classes at the bottom of the test class